components/ide/jetbrains/gateway-plugin/src/main/kotlin/io/gitpod/jetbrains/gateway/GitpodConnectionProvider.kt

```diff
@@ -230,6 +244,12 @@ class GitpodConnectionProvider : GatewayConnectionProvider {
                                         setErrorMessage("failed to fetch JetBrains Gateway Join Link.")
                                         return@launch
                                     }
+
+                                    // TODO: is this the right place?
```

Comment:
It should happen in the check above, ie in one transaction

Reply:
@akosyakov could you clarify what you mean for `transaction` in this context?

Reply:
Thanks @felladrin for clarifying in our sync chat, I think I understand what @akosyakov meant and I've updated the code accordingly. 🙏